provider/resources.go

```diff
@@ -550,15 +572,40 @@ func validateCredentials(vars resource.PropertyMap, c shim.ResourceConfig) error
 	}
 
 	if details, ok := vars["assumeRole"]; ok {
-
 		assumeRole := awsbase.AssumeRole{
-			RoleARN:     stringValue(details.ObjectValue(), "roleArn", []string{}),
-			ExternalID:  stringValue(details.ObjectValue(), "externalId", []string{}),
-			Policy:      stringValue(details.ObjectValue(), "policy", []string{}),
-			SessionName: stringValue(details.ObjectValue(), "sessionName", []string{}),
+			RoleARN:           stringValue(details.ObjectValue(), "roleArn", []string{}),
+			ExternalID:        stringValue(details.ObjectValue(), "externalId", []string{}),
+			Policy:            stringValue(details.ObjectValue(), "policy", []string{}),
+			PolicyARNs:        arrayValue(details.ObjectValue(), "policyArns", []string{}),
+			SessionName:       stringValue(details.ObjectValue(), "sessionName", []string{}),
+			SourceIdentity:    stringValue(details.ObjectValue(), "sourceIdentity", []string{}),
+			TransitiveTagKeys: arrayValue(details.ObjectValue(), "transitiveTagKeys", []string{}),
+		}
+		duration, err := durationFromConfig(details.ObjectValue(), "durationSeconds", []string{})
+		if err != nil {
+			return err
 		}
+		assumeRole.Duration = duration
+
 		config.AssumeRole = &assumeRole
+	}
+
+	if details, ok := vars["assumeRoleWithWebIdentity"]; ok {
+		assumeRole := awsbase.AssumeRoleWithWebIdentity{
+			RoleARN:              stringValue(details.ObjectValue(), "roleArn", []string{}),
+			Policy:               stringValue(details.ObjectValue(), "policy", []string{}),
+			PolicyARNs:           arrayValue(details.ObjectValue(), "policyArns", []string{}),
+			SessionName:          stringValue(details.ObjectValue(), "sessionName", []string{}),
+			WebIdentityToken:     stringValue(details.ObjectValue(), "webIdentityToken", []string{}),
+			WebIdentityTokenFile: stringValue(details.ObjectValue(), "webIdentityTokenFile", []string{}),
+		}
+		duration, err := durationFromConfig(details.ObjectValue(), "durationSeconds", []string{})
+		if err != nil {
+			return err
+		}
+		assumeRole.Duration = duration
 
+		config.AssumeRoleWithWebIdentity = &assumeRole
```

Comment:
I am very confused here because reading this code prior to the change it appeared that `config` only lives long enough to satisfy 'validateCredentials'. Which in turn is only used to make sure we can connect to AWS, but is not actually used to connect to AWS (which in itself is of course very suspect and error-prone, that the two methods diverge).

Are you intentional about this? That is does your change target fixing credentials validation change only without intending to change the actual behavior of the provider past the validation change? Or perhaps I'm misunderstanding how this works?

Reply:
Ah yes.. It sounds like that's what the ticket is asking for literally.

Reply:
No it's on me, should have read the ticket!